### PR TITLE
[PROTOTYPE] Timeline Segment Streaming

### PR DESCRIPTION
**Pull request by skovati**
* **Tickets addressed:** AERIE-1921
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description

Very experimental draft PR to explore timeline segment streaming options. Essentially in pre-draft state right now, but opening up the PR for feedback!

### CODE REVIEW

File: merlin-server/sql/merlin/tables/lifecycle_event.sql [hunk 1]
```diff
@@ -0,0 +1,35 @@
+CREATE TYPE event_t AS ENUM ('begin', 'end', 'wait');
```
Comment (Twisol) [inline merlin-server/sql/merlin/tables/lifecycle_event.sql]: I know the PR is essentially "pre-draft", but FWIW, most if not all of the other SQL files use lowercase keywords. (I've always found SQL kind of shouty, and I think good formatting tends to bring most of the benefits that uppercase keywords do anyway.)

Comment (skovati) [reply merlin-server/sql/merlin/tables/lifecycle_event.sql]: Good call! CHANGE HAS BEEN MADE

File: merlin-server/sql/merlin/tables/lifecycle_event.sql [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        references activity (id)
+        on update cascade
+        on delete cascade
+);
+
+comment on table lifecycle_event is e''
+    'a lifecycle event for a specific activity meant to be streamed out';
+
+comment on column lifecycle_event.timeline_segment_id is e''
+    'the timeline_segment that this even belongs to';
```
Comment (dyst5422) [inline merlin-server/sql/merlin/tables/lifecycle_event.sql]: typo

**Comment by camargo** [comment]
Closing this for the time being until we can revisit

**Pull request closed without merging**